### PR TITLE
Add thresholds to CPA analyzers.

### PR DESCRIPTION
**Pull request by Cameron-Micka**
This is a customer request to add discreate thresholds (if they exist) to each CPA analyzer.

### CODE REVIEW

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 1]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
-- **Rigidbody Complexity** (rigidbody counts)
-- **Scene Complexity** (batches and draw calls)
-- **Shader Complexity** (vertex and fragment shader math instruction counts, variant counts)
-- **SRP Batcher Compatible** (compatibility check with the SRP batcher)
-- **Terrain** (terrain component settings)
-- **Texture Sizes** (memory usage for lightmaps and other textures)
-- **WebSlate** (runtime render time for WebSlate)
-
-**Scene Complexity** and **Mesh Complexity** results are based on camera position and orientation. The CPA tool will search for all cameras within your scene, including disabled ones, and run tests from each camera. If your scene doesn't have a camera, the CPA tool will add a camera when testing that focuses on the bounds of your scene.
+Each analyzer performs a different test to check if content is optimally setup. The below table highlights what each analyzer does as well as discrete thresholds.
```
Comment (vtieto) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
Each analyzer performs a different test to check if content is optimally setup. The table below highlights what each analyzer does as well as discrete thresholds.
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 2]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+
+| Name | Description | Requires Play Mode | PC thresholds | Android (Quest) thresholds |
+| -------- | ------- | ------- | ------- | ------- |
+| **Animation** | Checks if Animator and Animation components have visibility culling setup optimally. | No | N/A | N/A |
```
Comment (vtieto) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Animation** | Checks if Animator and Animation components have visibility culling set up optimally. | No | N/A | N/A |
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 3]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+| **Collision Complexity** | Checks for high triangle count MeshCollider components. | No | Under 100 triangles for dynamic MeshColliders and under 10,000 triangles for static MeshColliders | Same as PC |
+| **Light Complexity** | Checks utilization of realtime lights and shadow casting lights. And for the existence of an avatar light. | No | Under 5 realtime per pixel lights. And 0 realtime shadow casting lights. | Under 3 realtime per pixel lights. And 0 realtime shadow casting lights. |
```
Comment (vtieto) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Light Complexity** | Checks utilization of real-time lights and shadow casting lights and for the existence of an avatar light. | No | Under 5 real-time per pixel lights and 0 real-time shadow casting lights. | Under 3 real-time per pixel lights and 0 real-time shadow casting lights. |
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 4]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+| **Mesh Complexity** | Warns about the use of very dense MeshFilter components. | No | Above 30,000 triangles per MeshFilter emits a warning and above 100,000 triangles per MeshFilter emits an error. | Same as PC |
+| **Mesh Complexity Rendered** | Checks if a viewpoint is rendering too many triangles. | Yes | Under 500,000 triangles rendered. | Under 80,000 triangles rendered. |
+| **Reflection Probes** | Checks if realtime reflection probes are being used. | No | Allowed | Disallowed |
```
Comment (vtieto) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Reflection Probes** | Checks if real-time reflection probes are being used. | No | Allowed | Disallowed |
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 5]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+| **Rigidbody Complexity** | Checks for too many Rigidbody components being used. | No | Under 50 rigid bodies. | Same as PC |
```
Comment (vtieto) [inline mesh/develop/debug-and-optimize-performance/cpa.md]: ### 
```suggestion
| **Rigidbody Complexity** | Checks for too many Rigidbody components being used. | No | Under 50 Rigidbodies. | Same as PC |
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 6]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+| **Scene Complexity** | Checks if a viewpoint is emitting too many rendering batches. | Yes | Under 200 batches emitted. | Under 50 batches emitted |
+| **Shader Complexity** | Checks the number vertex and fragment stage math operations of referenced shaders. | No | Under 150 math operations for the vertex stage. Under 600 math operations for the fragment stage. | Under 30 math operations for the vertex stage. Under 120 math operations for the fragment stage. |
```
Comment (vtieto) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Shader Complexity** | Checks the number of vertex and fragment stage math operations of referenced shaders. | No | Under 150 math operations for the vertex stage. Under 600 math operations for the fragment stage. | Under 30 math operations for the vertex stage. Under 120 math operations for the fragment stage. |
```

**Review by vtieto — APPROVED** [review]

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 7]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
-- **Texture Sizes** (memory usage for lightmaps and other textures)
-- **WebSlate** (runtime render time for WebSlate)
-
-**Scene Complexity** and **Mesh Complexity** results are based on camera position and orientation. The CPA tool will search for all cameras within your scene, including disabled ones, and run tests from each camera. If your scene doesn't have a camera, the CPA tool will add a camera when testing that focuses on the bounds of your scene.
+Each analyzer performs a different test to check if content is optimally setup. The table below highlights what each analyzer does as well as discrete thresholds.
+
+| Name | Description | Requires Play Mode | PC thresholds | Android (Quest) thresholds |
+| -------- | ------- | ------- | ------- | ------- |
+| **Animation** | Checks if Animator and Animation components have visibility culling set up optimally. | No | N/A | N/A |
+| **Collision Complexity** | Checks for high triangle count MeshCollider components. | No | Under 100 triangles for dynamic MeshColliders and under 10,000 triangles for static MeshColliders | Same as PC |
```
Comment (Cameron-Micka) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Collision Complexity** | Checks for high triangle count MeshCollider components. | No | Under 100 triangles for dynamic MeshColliders and under 10,000 triangles for static MeshColliders. | Same as PC |
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 8]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+| **Light Complexity** | Checks utilization of real-time lights and shadow casting lights and for the existence of an avatar light. | No | Under 5 real-time per pixel lights and 0 real-time shadow casting lights. | Under 3 real-time per pixel lights and 0 real-time shadow casting lights. |
+| **Mesh Complexity** | Warns about the use of very dense MeshFilter components. | No | Above 30,000 triangles per MeshFilter emits a warning and above 100,000 triangles per MeshFilter emits an error. | Same as PC |
+| **Mesh Complexity Rendered** | Checks if a viewpoint is rendering too many triangles. | Yes | Under 500,000 triangles rendered. | Under 80,000 triangles rendered. |
+| **Reflection Probes** | Checks if real-time reflection probes are being used. | No | Allowed | Disallowed |
+| **Rigidbody Complexity** | Checks for too many Rigidbody components being used. | No | Under 50 Rigidbodies. | Same as PC |
+| **Scene Complexity** | Checks if a viewpoint is emitting too many rendering batches. | Yes | Under 200 batches emitted. | Under 50 batches emitted |
```
Comment (Cameron-Micka) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Scene Complexity** | Checks if a viewpoint is emitting too many rendering batches. | Yes | Under 200 batches emitted. | Under 50 batches emitted. |
```

File: mesh/develop/debug-and-optimize-performance/cpa.md [hunk 9]
```diff
@@ -34,20 +34,25 @@ Click the **Select Platform Quality** drop-down and then select your desired pla
...
+| **Mesh Complexity Rendered** | Checks if a viewpoint is rendering too many triangles. | Yes | Under 500,000 triangles rendered. | Under 80,000 triangles rendered. |
+| **Reflection Probes** | Checks if real-time reflection probes are being used. | No | Allowed | Disallowed |
+| **Rigidbody Complexity** | Checks for too many Rigidbody components being used. | No | Under 50 Rigidbodies. | Same as PC |
+| **Scene Complexity** | Checks if a viewpoint is emitting too many rendering batches. | Yes | Under 200 batches emitted. | Under 50 batches emitted. |
+| **Shader Complexity** | Checks the number of vertex and fragment stage math operations of referenced shaders. | No | Under 150 math operations for the vertex stage. Under 600 math operations for the fragment stage. | Under 30 math operations for the vertex stage. Under 120 math operations for the fragment stage. |
+| **SRP Batcher Compatible** | Checks if SRP Batcher compatible shaders are being used. | No | N/A | N/A |
+| **Terrain** | Checks for performance issues on Terrain components. | No | Under 2 Terrain components and a Heightmap Pixel Error over 4 | Same as PC |
```
Comment (Cameron-Micka) [inline mesh/develop/debug-and-optimize-performance/cpa.md]:
```suggestion
| **Terrain** | Checks for performance issues on Terrain components. | No | Under 2 Terrain components and a Heightmap Pixel Error over 4. | Same as PC |
```

**Pull request merged**